src/Plugins/Convert.php

```diff
@@ -0,0 +1,932 @@
+            $amount = (int) $firstParamArr[0];
+            $paramsArr[0] = $firstParamArr[1];
+    
+        } else {
+
+            $amount = 1;
+
+        }
+
+        if ( ( $toPosition = array_search('to', $paramsArr) ) === false) {
+
+            throw new InvalidParametersException("The from unit type and to unit type do not match.");
+
+        }
+    
+        $fromUnit = implode(" ", array_slice($paramsArr, 0, $toPosition));
+        $toUnit = implode(" ", array_slice($paramsArr, $toPosition + 1));
+
+        return [
+            "amount" => $amount,
+            "from_unit" => $fromUnit,
+            "to_unit" => $toUnit
+        ];
+    }
+
+    public function convert(Command $command): Promise
+    {
+        $paramsArr = $command->getParameters();
+
+        try {
+
+            $parsedParams = $this->parseParams($paramsArr);
+
+        } catch (\InvalidParametersException $e) {
+
+            return $this->chatClient->postReply(
+                $command,
+                "I'm really not sure what you want me to do..."
+            );
+
+        }
+
+        if ($parsedParams['amount'] === 0) {
+
+            return $this->chatClient->postReply(
+                $command,
+                "Yeah let me just divide this by 0 and..."
+            );
+
+        }
+
+        try {
+
+            $fromUnitArr = $this->getUnitType($parsedParams['from_unit']);
+            $toUnitArr = $this->getUnitType($parsedParams['to_unit']);
+
+        } catch (\InvalidParametersException $e) {
+
+            return $this->chatClient->postReply(
+                $command,
+                "They didn't teach me that one yet."
+            );
+
+        }
+
+        if ( ( $fromType = $fromUnitArr[0] ) !== ( $toType = $toUnitArr[0] ) ) {
+
+            return $this->chatClient->postReply(
+                $command, 
+                "What?? They never taught me how to convert " . $fromType . " to " . $toType . "."
+            );
+
+        }
+
+        switch($fromType) {
+            case 'temperature':
+                $convertedToAmount = $this->tempConvert(
+                    $fromUnitArr[1], 
+                    $toUnitArr[1], 
+                    $parsedParams['amount']
+                );
+                break;
+
+            case 'currency':
+                $parsedParams['amount'] = number_format($parsedParams['amount'], 2, '.', ',');
+
+                try {
+
+                    $convertedCurrency = $this->currencyConvert(
+                        $fromUnitArr[1],
+                        $toUnitArr[1],
+                        $parsedParams['amount']
+                    );
+
+                } catch (\RequestFailedException $e) {
+
+                    $message = "I don't know what's going on but I can't find that information.";
+
+                    return $this->chatClient->postReply(
+                        $command, 
+                        $message
+                    );
+
+                }
```

Comment:
It would be better if the body of the `case` is moved to a method. So you only have a method call in here.